python/tskit/trees.py

```diff
@@ -5935,6 +5935,111 @@ def trim(self, record_provenance=True):
+        ``node_time[child] < time < node_time[parent]`` with two edges
+        ``(left, right, parent, u)`` and ``(left, right, u, child)``,
+        where ``u`` is a newly added node for each intersecting edge. If
+        ``metadata`` or ``population`` are specified, newly added nodes will be
+        assigned these values. Otherwise, default values will be used. If a
+        metadata schema is defined for the node table, the empty dictionary
+        will be used by default for the new node when calling
+        :meth:`.NodeTable.add_row`; otherwise, an empty byte string.
+
+        Any metadata associated with the edge will be copied to the new edge.
+
+        The population value for the new node will be derived from the
+        population of the edge's child.
+
+        .. warning:: This method currently does not support migrations
+            and a ValueError will be raised if the migration table is not
+            empty. Future versions may take migrations that intersect with the
+            edge into account when determining the default population
+            assignments for new nodes.
+
+        Newly added nodes will have a ``flags`` value of 0, if not specified.
+
+        Any mutations lying on the edge whose time is >= ``time`` will have
+        their node value set to ``u``. Note that the time of the mutation is
+        defined as the time of the child node if the mutation's time is
+        unknown.
+
+        :param float time: The cutoff time.
+        :param int flags: The flags value for newly-inserted nodes. (Default = 0)
+        :param int population: The population value for newly inserted nodes.
+            Defaults to the population of the child node of the split edge
+            if not specified.
+        :param metadata: The metadata for any newly inserted nodes. See
+            :meth:`.NodeTable.add_row` for details on how default metadata
+            is produced for a given schema (or none).
+        :param bool record_provenance: If True, add details of this operation to the
+            provenance information of the returned tree sequence. (Default: True).
+        :rtype: tskit.TreeSequence
+        """
+        tables = self.dump_tables()
+        if self.num_migrations > 0:
+            raise ValueError("Migrations not supported")
+        default_population = population is None
+        if not default_population:
+            # -1 is a valid value
+            if population < -1 or population >= self.num_populations:
+                raise ValueError("Population out of bounds")
+        flags = 0 if flags is None else flags
+        if metadata is None:
+            metadata = tables.nodes.metadata_schema.empty_value
+        metadata = tables.nodes.metadata_schema.validate_and_encode_row(metadata)
+        # This is the easiest way to turn off encoding when calling add_row below
+        schema = tables.nodes.metadata_schema
+        tables.nodes.metadata_schema = tskit.MetadataSchema(None)
```

Comment:
Why do you do this?

Reply:
It was the only way I could figure out to get the defaults for metadata to work :cry:

Reply:
I feel your pain.